test/hotspot/jtreg/runtime/exceptionMsgs/NoClassDefFoundError/libNoClassDefFoundErrorTest.c

```diff
@@ -42,3 +46,34 @@ Java_NoClassDefFoundErrorTest_callFindClass(JNIEnv *env, jclass klass, jstring c
 }
 
 
+static char* giant_string() {
+    size_t len = ((size_t)INT_MAX) + 3;
+    char* c_name = malloc(len * sizeof(char));
+    if (c_name != NULL) {
+        memset(c_name, 0x59595959, len - 1); // YYYY...
```

Comment:
nit, `59` should be enough

Reply:
?? you mean just 0x59? That would be zero extended to 0x00000059, which would not have the desired affect.

Reply:
memset does cast the input int to unsigned char.

Reply:
Hmmm I was getting weird behaviour when I directly used the character e.g. `Y` - I attributed it to zero extension. I will check again.